src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -592,7 +594,30 @@ private void startAgent(SlaveComputer computer, final TaskListener listener, Str
                             String workingDirectory) throws IOException {
         session = connection.openSession();
         expandChannelBufferSize(session,listener);
-        String cmd = "cd \"" + workingDirectory + "\" && " + java + " " + getJvmOptions() + " -jar " + AGENT_JAR +
+
+        PrintStream logger = listener.getLogger();
+
+         // Code to support alternate drives (such as D:\) for the remote root directory on Windows, when not using cygwin.
+         // You can disable this feature with startup option: -Dhudson.plugins.sshslaves.alternativewindowsdrive=false
+        boolean AlternativeWindowsDriveCheck = ! "false".equals(System.getProperty("hudson.plugins.sshslaves.alternativewindowsdrive"));
+        String agentOSTYPE="";
+        try {
+            ByteArrayOutputStream agentOsTypeStream = new ByteArrayOutputStream();
+            connection.exec("echo \"$OSTYPE\"", agentOsTypeStream);
```

Comment:
Is this value going to differ from the system property value for `os.arch` or the others? That could be simpler than checking the shell like that, though this approach would still work.

Reply:
Is `os.arch` for the agent available when the remoting connection is just established (at this point of the code)? I had first tried checking properties, but I had found that they were either for the Jenkins Master (not helpful), or they were not initialized for the Jenkins agent yet, hence why I went this route.

Reply:
@jvz I've addressed all the other comments except this one